tests/ErrorHandlerTest.php

```diff
@@ -6,153 +6,52 @@
-
-            restore_error_handler();
-
-            $this->assertSame([$errorHandler, 'handleError'], $previousErrorHandler);
-        } finally {
-            restore_error_handler();
-            restore_exception_handler();
-        }
-    }
-
-    /**
-     * @dataProvider constructorThrowsWhenReservedMemorySizeIsWrongDataProvider
-     *
-     * @expectedException \UnexpectedValueException
-     * @expectedExceptionMessage The $reservedMemorySize argument must be greater than 0.
-     */
-    public function testConstructorThrowsWhenReservedMemorySizeIsWrong(int $reservedMemorySize): void
-    {
-        ErrorHandler::register($this->callbackMock, $reservedMemorySize);
-    }
-
-    public function constructorThrowsWhenReservedMemorySizeIsWrongDataProvider(): array
-    {
-        return [
-            [-1],
-            [0],
-        ];
-    }
-
-    /**
-     * @dataProvider handleErrorShouldNotCaptureDataProvider
-     */
-    public function testHandleErrorShouldNotCapture(bool $expectedToCapture, int $captureAt): void
-    {
-        if (!$expectedToCapture) {
-            $this->callbackMock->expects($this->never())
-                ->method('__invoke');
-        }
-
-        $errorHandler = ErrorHandler::register($this->callbackMock);
-        $errorHandler->captureAt($captureAt, true);
-
-        $prevErrorReporting = error_reporting(E_ERROR); // to avoid making the test error bubble up and make the test fail
-
-        try {
-            $this->assertFalse($errorHandler->handleError(E_WARNING, 'Test', __FILE__, __LINE__));
-        } finally {
-            error_reporting($prevErrorReporting);
-            restore_error_handler();
-            restore_exception_handler();
-            $this->addToAssertionCount(1);
-        }
-    }
-
-    public function handleErrorShouldNotCaptureDataProvider(): array
-    {
-        return [
-            [false, E_ERROR],
-            [true, E_ALL],
-        ];
-    }
-
-    /**
-     * @dataProvider captureAtDataProvider
-     */
-    public function testCaptureAt($levels, $replace, $expectedCapturedErrors): void
+    public function testGetCurrent(): void
     {
-        try {
-            $errorHandler = ErrorHandler::register($this->callbackMock);
-            $previousCapturedErrors = $this->getObjectAttribute($errorHandler, 'capturedErrors');
-
-            $this->assertEquals($previousCapturedErrors, $errorHandler->captureAt($levels, $replace));
-            $this->assertAttributeEquals($expectedCapturedErrors, 'capturedErrors', $errorHandler);
-        } finally {
-            restore_error_handler();
-            restore_exception_handler();
-        }
-    }
+        $errorHandler = ErrorHandler::registerOnce();
 
-    public function captureAtDataProvider(): array
-    {
-        return [
-            [E_USER_NOTICE, false, E_ALL],
-            [E_USER_NOTICE, true, E_USER_NOTICE],
-        ];
+        $this->assertSame($errorHandler, ErrorHandler::registerOnce());
```

Comment:
You can just inline the two calls to `ErrorHandler::registerOnce()`, we're not interested in reusing the instance so it makes no sense to store it into a local variable